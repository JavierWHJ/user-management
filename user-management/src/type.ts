export interface User {
    id: number,
    firstName: string,
    lastName: string,
    email: string,
    dob: number
}

export interface UserData {
    firstName: string,
    lastName: string,
    email: string,
    dob: string
}