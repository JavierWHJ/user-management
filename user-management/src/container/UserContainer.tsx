import React from "react";
import { Alert, Button } from "react-bootstrap";
import { Table } from "react-bootstrap";
import SearchComponent from "../components/SearchComponent";
import UserComponent from "../components/UserComponent";
import { User } from "../type";

interface Props {
    users: User[] | undefined;
}

const UserContainer = (props: Props) => {
    return (
        <div>
            <div className="mt-4 d-flex justify-content-between">
                <h2>Users</h2>
                <Button>Add User</Button>
            </div>
            <SearchComponent />
            {props.users && props.users.length !== 0 ? (
                <Table
                    striped
                    bordered
                    hover
                    responsive
                    style={{
                        marginTop: "20px",
                    }}
                >
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Email</th>
                            <th>Date of Birth</th>
                            <th>Actions</th>
                        </tr>
                    </thead>
                    <tbody>
                        {props.users
                            ? props.users.map((user) => {
                                    return (
                                        <UserComponent
                                            key={user.id}
                                            user={user}
                                        />
                                    );
                            })
                            : ""}
                    </tbody>
                </Table>
            ) : (
                <Alert
                    variant="info"
                    style={{
                        marginTop: "20px",
                    }}
                >
                    {"There are no users created yet."}
                </Alert>
            )}
        </div>
    );
};

export default UserContainer;
