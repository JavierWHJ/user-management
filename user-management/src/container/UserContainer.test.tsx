import React from "react";
import UserContainer from "./UserContainer";

import { render } from "@testing-library/react";
import { UserData } from "../type";

const users = [
    {
      firstName: "Bob",
      lastName: "Doe",
      email: "bob.doe@gmail.com",
      dob: 785635200000,
      id: 3
    }
];

const addUser = (newUser: UserData) => {
}

const updateUser = (id: number, userToUpdate: UserData) => {
}

const deleteUser = (id: number) => {
}

const searchUser = (query: string) => {
};

it("matches snapshot", () => {
  const { asFragment } = render(<UserContainer users={users} searchUser={searchUser} addUser={addUser} updateUser={updateUser} deleteUser={deleteUser} />);

  expect(asFragment()).toMatchSnapshot();
});