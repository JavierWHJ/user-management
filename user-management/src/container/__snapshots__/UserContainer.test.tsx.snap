// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mt-4 d-flex justify-content-between"
    >
      <h2>
        Users
      </h2>
      <button
        class="btn btn-primary"
        type="button"
      >
        Add User
      </button>
    </div>
    <div>
      <input
        class="form-control form-control-lg mt-3"
        name="searchterm"
        placeholder="Search User"
        type="text"
      />
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped table-bordered table-hover"
        style="margin-top: 20px;"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Date of Birth
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <div
                class="d-flex align-items-center"
              >
                <div
                  class="avatar-circle"
                >
                  <span
                    class="initials"
                  >
                    BD
                  </span>
                </div>
                <span
                  class="ml-3"
                >
                  Bob Doe
                </span>
              </div>
            </td>
            <td>
              bob.doe@gmail.com
            </td>
            <td>
              24 Nov 1994
            </td>
            <td>
              <div
                class="d-flex justify-content-center mt-2"
              >
                <button
                  class="mr-4 btn btn-success btn-block"
                  type="button"
                >
                  Update
                </button>
                <button
                  class="mt-0 btn btn-danger btn-block"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
